test/Algebra/Graph/Test/GraphRules.hs

```diff
@@ -0,0 +1,99 @@
+{-# LANGUAGE TemplateHaskell #-}
+-----------------------------------------------------------------------------
+-- |
+-- Module     : Algebra.Graph.Test.Graph
+-- Copyright  : (c) Andrey Mokhov 2016-2018
+-- License    : MIT (see the file LICENSE)
+-- Maintainer : andrey.mokhov@gmail.com
+-- Stability  : experimental
+--
+-- Testsuite for "Algebra.Graph" rewrite-rules
+-----------------------------------------------------------------------------
+module Algebra.Graph.Test.GraphRules where
+
+import Data.Maybe (fromMaybe)
+
+import Algebra.Graph
+import Algebra.Graph.Internal
+
+import qualified Test.Inspection as I
+
+-- overlays/connects tests
+--- overlays . map vertex
+vertices', overlaysDotMapVertex :: [a] -> Graph a
+vertices'            = fromMaybe Empty . foldr (mf Overlay . Vertex) Nothing
```

Comment:
Why are you testing a function defined here instead of the actual library `vertices`?

Reply:
Same for `clique`, `empty`, etc.

Reply:
Because otherwise the tests are failing :) It seems that with the declaration `f = g`, `g` is not inlined, even if it has an `INLINE` pragma...

Reply:
I think this is because we are working outside the module itself, but I find a bit messy to have these tests included directly in `src/Algebra/Graph.hs`...